src/isomorphic/classic/class/ReactClass.js

```diff
@@ -437,6 +437,21 @@ function validateMethodOverride(isAlreadyDefined, name) {
  */
 function mixSpecIntoComponent(Constructor, spec) {
   if (!spec) {
```

Comment:
@gaearon Since we have this condition, a mixin in case if is an array, will never get inside this block. 
I feel if a mixin is an array, then its better we show a separate warning that says: 
```
Warning: Your\'re attempting to include a mixin that is an array. Expected object but got array
```